### PR TITLE
Allow weight and height to use metric units.

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
Also fixed a bug where you could change the height but because you are not in edit-mode the change is not saved.

Note: Ideally this would be updated to use one pattern for all the user data, but that would have required more time.

### CODE REVIEW

File: APCAppCore/APCAppCore/DataSubstrate/Model/APCUser+UserData.m [hunk 1]
```diff
@@ -127,6 +128,91 @@ + (NSArray *) medications {
...
+    NSLengthFormatterUnit formatterUnit;
+    NSLengthFormatter *formatter = [[NSLengthFormatter alloc] init];
+    formatter.unitStyle = NSFormattingUnitStyleMedium;
+    formatter.forPersonHeightUse = YES;
+    [formatter unitStringFromMeters:2.0 usedUnit:&formatterUnit];
+    
+    // Convert the formatter unit to a HKUnit (default to centimeter)
+    NSArray *hkUnits = @[[HKUnit meterUnitWithMetricPrefix:HKMetricPrefixCenti]];
+    NSArray *formatterUnits = @[@(NSLengthFormatterUnitCentimeter)];
+    NSArray *maxValues = @[@(244)];
```
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/DataSubstrate/Model/APCUser+UserData.m]: 8'11'' ~ 272cm

Comment (syoung-smallwisdom) [inline APCAppCore/APCAppCore/DataSubstrate/Model/APCUser+UserData.m]: True. Are there really people over 8' tall?

Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/DataSubstrate/Model/APCUser+UserData.m]: https://en.wikipedia.org/wiki/List_of_tallest_people

Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/DataSubstrate/Model/APCUser+UserData.m]: (OK so only 4 of them are currently alive)

File: APCAppCore/APCAppCore/DataSubstrate/Model/APCUser+UserData.m [hunk 2]
```diff
@@ -127,6 +128,91 @@ + (NSArray *) medications {
...
+    NSLengthFormatterUnit formatterUnit;
+    NSLengthFormatter *formatter = [[NSLengthFormatter alloc] init];
+    formatter.unitStyle = NSFormattingUnitStyleMedium;
+    formatter.forPersonHeightUse = YES;
+    [formatter unitStringFromMeters:2.0 usedUnit:&formatterUnit];
+    
+    // Convert the formatter unit to a HKUnit (default to centimeter)
+    NSArray *hkUnits = @[[HKUnit meterUnitWithMetricPrefix:HKMetricPrefixCenti]];
+    NSArray *formatterUnits = @[@(NSLengthFormatterUnitCentimeter)];
+    NSArray *maxValues = @[@(floor((8 * 12 + 11) * 25.4))];
```
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/DataSubstrate/Model/APCUser+UserData.m]: 2.54

**Comment by Erin-Mounts** [comment]
looks good.

**Pull request merged**